### PR TITLE
Feature/relationships selection process process stage

### LINKED ISSUE #39: Relationships SelectionProcess-ProcessStage

### PR DESCRIPTION
**Pull request by ivancolomer**
Added OneToMany and OneToOne relatioships between selectionProcess and processStage

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/academicrecruit/domain/ProcessStage.java [hunk 1]
```diff
@@ -30,4 +33,12 @@
...
     private Date beginDate;
     private Date endDate;
+
+    @ManyToOne
+    @JsonIdentityReference(alwaysAsId = true)
+    private SelectionProcess selectionProcess;
+
+    @OneToOne(mappedBy = "activeProcessStage")
+    @JsonIdentityReference(alwaysAsId = true)
+    private SelectionProcess selectionProcessBeingActive;
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/academicrecruit/domain/ProcessStage.java]: Remember we said we would only set the relation on the dominating side of the relation. You should remove the relation from this side.

File: src/main/java/cat/udl/eps/softarch/academicrecruit/domain/SelectionProcess.java [hunk 2]
```diff
@@ -26,4 +30,14 @@
     @NotBlank
     @Length(max = 256)
     private String vacancy;
+
+    @OneToMany (mappedBy = "selectionProcess")
+    @JsonIdentityReference(alwaysAsId = true)
+    private List<ProcessStage> processStages;
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/academicrecruit/domain/SelectionProcess.java]: Same here, you should remove the relation from this side.

**Review by bergacat1 — CHANGES_REQUESTED** [review]

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateProcessStageStepDefs.java [hunk 3]
```diff
@@ -49,4 +55,32 @@ public void thePreviouslyUpdatedProcessStageHasNowStep(int newStep) throws Throw
                 .andDo(print())
                 .andExpect(jsonPath("$.step", is(newStep)));
     }
+
+    @Transactional
```
Comment (bergacat1) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateProcessStageStepDefs.java]: Not necessary now

**Review by bergacat1 — DISMISSED** [review]

**Comment by rogargon** [comment]
@paulavicente @ivancolomer Do not forget to address the last comment from Albert so this PR can be accepted

**Review by rogargon — APPROVED** [review]

**Pull request merged**